nestjs/src/courses/team-distribution/dto/team.dto.ts

```diff
@@ -0,0 +1,60 @@
+import { Team } from '@entities/team';
+import { ApiProperty } from '@nestjs/swagger';
+import { PaginationMeta } from 'src/core/paginate';
+import { PaginationMetaDto } from 'src/core/paginate/dto/Paginate.dto';
+import { TeamDistributionStudentDto } from './team-distribution-student.dto';
+
+export class TeamPasswordDto {
+  constructor(team: Team) {
+    this.password = `${team.id}_${team.password}`;
+  }
+
+  @ApiProperty()
+  public password: string;
+}
+
+export class TeamDto {
+  constructor(team: Team) {
+    this.id = team.id;
+    this.name = team.name;
+    this.chatLink = team.chatLink;
+    this.description = team.description;
+    this.students = team.students.map(st => new TeamDistributionStudentDto(st));
+    this.teamLeadId = team.teamLeadId;
+    this.teamDistributionId = team.teamDistributionId;
+  }
+
+  @ApiProperty()
+  public id: number;
+
+  @ApiProperty()
+  public name: string;
+
+  @ApiProperty()
+  public chatLink: string;
+
+  @ApiProperty()
+  public description: string;
+
+  @ApiProperty()
```

Comment:
to generate type properly should do a trick:
`@ApiProperty({ type: [TeamDistributionStudentDto] })`

Reply:
Dziękuję